agent/service.go

```diff
@@ -253,6 +253,9 @@ func (ts *TaskService) Create(requestCtx context.Context, req *taskAPI.CreateTas
 			logger.WithError(err).Error()
 			return nil, err
 		}
+		ts.addCleanup(taskExecID, func() error {
+			return fifoSet.Close()
```

Comment:
These are fifos inside the VM, right? Once the task gets deleted, the VM should be shut down, so how is this contributing to the leak?

Reply:
> Once the task gets deleted, the VM should be shut down

The VM _can_ shutdown when all tasks are stopped, but that is not always the case. Default is for VM to stay up even if all tasks are deleted. See https://github.com/firecracker-microvm/firecracker-containerd/blob/main/proto/firecracker.proto#L34

However, you are correct `Create` cleanup removes the top level directory here and wouldn't contribute to the leak. Making this not _required_ but doesn't hurt anything to have either. Would you prefer to exclude this?

Reply:
Okay, just wanted to understand if this was contributing to the leak as well besides the exec path. I am fine keeping this in as an extra safety measure.